### PR TITLE
Fixes compile warning introduced in #25987

### LINKED ISSUE #26239: Fix newly-introduced compiler warning in AsyncDnsResolver.scala
Affects #26088. This is new since the akka-actor compiler warnings were fixed, with the exception of only those few needing to retain. I see no reason for this to be retained in addition, it just needs a default. Ideally we automate this, I found it while starting #26087

````
akka/akka-actor/src/main/scala/akka/io/dns/internal/AsyncDnsResolver.scala:122:5: match may not be exhaustive.
[warn] It would fail on the following input: Nil
[warn]     searchNames match {
[warn]     ^
```

### PR DESCRIPTION
**Pull request by jroper**
Fixes #26239

### CODE REVIEW

**Comment by jroper** [comment]
@helena Heres the fixed compile warning.

File: akka-actor/src/main/scala/akka/io/dns/internal/AsyncDnsResolver.scala [hunk 1]
```diff
@@ -127,6 +127,9 @@ private[io] final class AsyncDnsResolver(
           if (resolved.records.isEmpty) resolveFirst(remaining, requestType, resolver)
           else Future.successful(resolved)
         }
+      case Nil ⇒
+        // This can't happen
+        sys.error("resolveFirst shouldn't be called with empty searchNames")
```
Comment (patriknw) [inline akka-actor/src/main/scala/akka/io/dns/internal/AsyncDnsResolver.scala]: I'd prefer `throw IllegalStateException` since may go noticed

**Comment by helena** [comment]
Thanks @jroper ! Looks like just a minor change in response to `Nil` requested.

**Review by helena — COMMENTED** [review]
Minor change for `Nil`

**Comment by helena** [comment]
Closing to not bother @jroper and the replacement fixes 2 new. This is the only manual fix before adding automated build checks. Thanks @jroper !

**Comment by jroper** [comment]
Sorry @helena, this slipped through the cracks. Thanks for following up.

**Pull request closed without merging**